### ISSUE TITLE
table presenter messes selection when sorting by column

### ISSUE THREAD

**Issue by demarey** [pharo-spec/Spec#842]
Currently, if you select an item in a table presenter, then click on a column header to sort by this column, there is an inconsistency between what is displayed and what is selected.

Table presenter selection only holds the selected index(es). It then ask the model for the item at the given index(es). But, when sorting is activated, the model is not aware of the sorting (only the datasource is) and then give you back the item at index just like if the items where not sorted.
To summarize, visual feedback is good : you feel like the right element is selected but if you ask the selectedItem(s), then you get different items!

**Comment by demarey** [comment]
Mostly solved in 8a8e9cef8eef4d739ae72729ab15795b39c35982
Missing selection reset and tests
